main/main.cpp

```diff
@@ -2530,9 +2530,9 @@ void Main::cleanup(bool p_force) {
 		memdelete(camera_server);
 	}
 
+	finalize_navigation_server();
 	OS::get_singleton()->finalize();
 	finalize_physics();
```

Comment:
Does it make a difference if it's before or after `OS::finalize`? If it fixes it in both cases, I'd just move it before `physics` as the most minimal change needed to fix the issue.

Reply:
That is what I am currently testing cause when only moved before physics it still crashes with visuals from NavigationMesh. In Godot 4 the OS is placed far lower so this is not an issue but in Godot 3 it is right before physics.